cli.js

```diff
@@ -0,0 +1,37 @@
+const Scorecard = require('./scorecard'); 
+const readline = require('readline');
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+scorecard = new Scorecard;
+
+const askUserIput = () => {
+  rl.question(`How many pins did you roll in frame ${scorecard.frames.length + 1}? Please enter two numbers for your first and second rolls, separated by a comma e.g. 2,3 If you bowled a strike, simply enter 10.\n`, (answer) => {
+  const rolls = answer.split(',');
+  frame = rolls.map(roll => {
+    return parseInt(roll); // turn string into number
+  });
+  scorecard.inputFrame(frame);
+  if (scorecard.frames.length > 9) { 
+    confirmOutcome(); 
+    return rl.close();
+  }
+  console.log('Thank you. Your rolls have been added to your scorecard. Your final score will be calculated after all 10 frames have been entered.\n');
+  askUserIput();
```

Comment:
Nice use of recursion. Something else you could have done is a while loop, potentially with a break. 
Typically a recursion can always be turned into a while loop, and vice versa.

Reply:
One is not better than the other. recursion is more "functional style", and sometimes easier to write, but can also be difficult to understand (her it's quite straightforward) and can lead to stack overflows.
While is a more "iterative style", and needs more state...
It's good to be aware of both and a very good skill to be able to convert one into the other and back :)